### ISSUE TITLE
Error when running Minishift in a VirtualBox VM (nested virtualization)

### ISSUE THREAD

**Issue by nextropia** [minishift/minishift#3119]
### General information

I'm getting the following error when trying to run Minishift inside an Ubuntu VM on VirtualBox. Nested hardware virtualization is enabled in VirtualBox. I tried running with a single vCPU (minishift start --cpus 1) but the error is the same. (More detailed libmachine log [here](https://gist.github.com/nextropia/f5e2cbf23e83446dee62c1694455f208))

```
vagrant@ubuntu-bionic:~$ minishift start
-- Starting profile 'minishift'
-- Check if deprecated options are used ... OK
-- Checking if https://github.com is reachable ... OK
-- Checking if requested OpenShift version 'v3.11.0' is valid ... OK
-- Checking if requested OpenShift version 'v3.11.0' is supported ... OK
-- Checking if requested hypervisor 'kvm' is supported on this platform ... OK
-- Checking if KVM driver is installed ...
   Driver is available at /usr/local/bin/docker-machine-driver-kvm ...
   Checking driver binary is executable ... OK
-- Checking if Libvirt is installed ... OK
-- Checking if Libvirt default network is present ... OK
-- Checking if Libvirt default network is active ... OK
-- Checking the ISO URL ... OK
-- Checking if provided oc flags are supported ... OK
-- Starting the OpenShift cluster using 'kvm' hypervisor ...
-- Starting Minishift VM .... FAIL E0111 11:35:25.467168    1657 start.go:494] Error starting the VM: Error getting the state for host: unexpected EOF. Retrying.
Error starting the VM: Error getting the state for host: unexpected EOF
```

Is it even possible to do this? 

  * Minishift version: 1.29.0
  * OS: Ubuntu 18.04 (virtualized)
  * Hypervisor: KVM

### Steps to reproduce
1. Run Vagrant VM on VirtualBox using this Vagrantfile:
https://github.com/nextropia/minishift-vagrant/blob/master/Vagrantfile

2. in the Vagrant VM: minishift start

### Logs
https://gist.github.com/nextropia/f5e2cbf23e83446dee62c1694455f208

**Comment by gbraad** [comment]
My general answer will be: 'Nested virtualization is not supported by us'.

Long answer, can you share the settings for `kvm_intel` ? Have you enabled `nested=1`?

**Comment by nextropia** [comment]
Thank you Gerard!

I did some further research and also tried Virtualbox 6.0 (instead of KVM), which apparently has improved support for nested virtualization, but I got a similar error. I've also found some information which suggests nested virtualization may be supported with Virtualbox only if the guest is 32bit, but then only in emulated mode and thus very slow.

My original goal was to create a Vagrant image I can easily share with my co-workers. In the end, the minishift executable is super easy to deploy and the actual Minishift "cluster" is provisioned in a VM anyway, so I don't think it's worth continuing down the path of packaging the whole thing as a complete image, especially since that's not supported. It's super easy to just run Minishift on the physical host.

Thanks again for looking into this!
